SmartDeviceLink/SDLProxy.m

```diff
@@ -715,6 +717,27 @@ - (void)handleSystemRequestLockScreenIconURL:(SDLOnSystemRequest *)request {
                 }];
 }
 
+- (void)sdl_handleSystemRequestIconURL:(SDLOnSystemRequest *)request {
+    __weak typeof(self) weakSelf = self;
+    [self sdl_sendDataTaskWithURL:[NSURL URLWithString:request.url]
+                completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+                    __strong typeof(weakSelf) strongSelf = weakSelf;
+                    if (error != nil) {
+                        SDLLogW(@"OnSystemRequest (icon url) HTTP download task failed: %@", error.localizedDescription);
+                        return;
+                    } else if (data.length == 0) {
+                        SDLLogW(@"OnSystemRequest (icon url) HTTP download task failed to get the cloud app icon image data");
+                        return;
+                    }
+
+                    SDLSystemRequest *iconURLSystemRequest = [[SDLSystemRequest alloc] initWithType:SDLRequestTypeIconURL fileName:request.url];
+                    iconURLSystemRequest.requestSubType = SDLRPCParameterNameId;
```

Comment:
Are you sure this is correct? I don't see it in the proposal.

Reply:
Removed. Checked with Core team.